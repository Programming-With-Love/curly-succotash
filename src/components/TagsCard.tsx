import * as React from 'react'
import { GatsbyLinkProps } from 'gatsby-link'

import { markdownRemarkGroupConnectionConnection } from '../graphql-types'

import * as classes from './TagsCard.module.scss'

export interface TagsCardProps extends React.HTMLProps<HTMLDivElement> {
  tags: string[]
  Link: React.ComponentClass<GatsbyLinkProps<any>>
  origin?: boolean
}

export default (props: TagsCardProps) => (
  <div className={classes.tagContainer}>
    {props.origin ? (
      <div title={'原创内容'} className={classes.original}>
        原
      </div>
    ) : null}
    {props.tags.map(tag => {
      const tagLink = `/blog/tags/${tag}/`
      return (
        <span key={tag} className={classes.tag}>
          <props.Link to={tagLink}>{tag}</props.Link>
        </span>
      )
    })}
  </div>
)
