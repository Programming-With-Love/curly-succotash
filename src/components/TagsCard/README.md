# TagsCard component

Component to have a pretty tags list from all posts.

## Source

    <TagsCard tags={props.data.tags} Link={Link}/>

## With tag property

    <TagsCard tags={props.data.tags} tag={tag} Link={Link}/>

## GraphQL query

    {
      # Get tags
      tags: allMarkdownRemark(frontmatter: {draft: {ne: true}}) {
        groupBy(field: frontmatter___tags) {
          fieldValue
          totalCount
        }
      }
    }

[open in graphiql](http://localhost:8000/graphql?query=%7B%0A%20%20%23%20Get%20tags%0A%20%20tags%3A%20allMarkdownRemark(frontmatter%3A%20%7Bdraft%3A%20%7Bne%3A%20true%7D%7D)%20%7B%0A%20%20%20%20groupBy(field%3A%20frontmatter___tags)%20%7B%0A%20%20%20%20%20%20fieldValue%0A%20%20%20%20%20%20totalCount%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D) (local only)
