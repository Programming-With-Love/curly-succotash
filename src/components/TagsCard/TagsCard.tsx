import * as React from 'react'
import { GatsbyLinkProps } from 'gatsby-link'
import { Card, List } from 'semantic-ui-react'
import { markdownRemarkGroupConnectionConnection } from '../../graphql-types'

interface TagsCardProps extends React.HTMLProps<HTMLDivElement> {
  tags: markdownRemarkGroupConnectionConnection[]
  Link: React.ComponentClass<GatsbyLinkProps<any>>
  tag?: string
}

export default (props: TagsCardProps) => {
  return (
    <Card>
      <Card.Content>
        <Card.Header>Tags</Card.Header>
      </Card.Content>
      <Card.Content>
        <List>
          {props.tags.map(tag => {
            const isActive = tag.fieldValue === props.tag
            const activeStyle = {
              fontWeight: '700',
            }
            const tagLink = isActive ? `/blog` : `/blog/tags/${tag.fieldValue}/`
            return (
              <List.Item as="span" key={tag.fieldValue}>
                <List.Icon name="tag" color={isActive ? 'blue' : null} />
                <List.Content style={isActive ? activeStyle : null}>
                  <props.Link to={tagLink}>
                    {tag.fieldValue} ({tag.totalCount})
                  </props.Link>
                </List.Content>
              </List.Item>
            )
          })}
        </List>
      </Card.Content>
    </Card>
  )
}
