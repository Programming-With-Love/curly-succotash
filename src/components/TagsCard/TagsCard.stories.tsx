const withReadme = (require('storybook-readme/with-readme') as any).default
const TagsCardReadme = require('./README.md')

import * as React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { withKnobs, select } from '@storybook/addon-knobs'
import TagsCard from './TagsCard'
import { markdownRemarkGroupConnectionConnection } from '../../graphql-types'

const tags = [
  { fieldValue: 'tag01', totalCount: 2 },
  { fieldValue: 'tag02', totalCount: 4 },
  { fieldValue: 'tag03', totalCount: 6 },
] as markdownRemarkGroupConnectionConnection[]

const LinkStub = ((props: any) => <div onClick={action(props.to.toString())}>{props.children}</div>) as any

storiesOf('TagsCard', module)
  .addDecorator(withReadme(TagsCardReadme))
  .addDecorator(withKnobs)
  .add('default', () => {
    return <TagsCard tags={tags} Link={LinkStub} />
  })
  .add('with tag property', () => {
    const tag = select('Tag', tags.map(t => t.fieldValue), 'tag01')
    return <TagsCard tags={tags} tag={tag} Link={LinkStub} />
  })
