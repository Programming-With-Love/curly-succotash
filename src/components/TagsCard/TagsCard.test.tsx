import { shallow, configure } from 'enzyme'
import 'jest'
import * as React from 'react'
import TagsCard from './TagsCard'

import { Card, List } from 'semantic-ui-react'
import { markdownRemarkGroupConnectionConnection } from '../../graphql-types'

// Configure enzyme with react 16 adapter
const Adapter: any = require('enzyme-adapter-react-16')
configure({ adapter: new Adapter() })

describe('TagsCard component', () => {
  let LinkStub: any

  beforeEach(() => {
    LinkStub = (props: any) => <div>{props.children}</div>
  })

  it('should list all the tags', () => {
    const tags = [
      { fieldValue: 'tag01', totalCount: 2 },
      { fieldValue: 'tag02', totalCount: 4 },
      { fieldValue: 'tag03', totalCount: 6 },
    ] as markdownRemarkGroupConnectionConnection[]

    const wrapper = shallow(<TagsCard tags={tags} Link={LinkStub} />)

    expect(wrapper.find(List.Item)).toHaveLength(3)
  })

  it('should have on tag active', () => {
    const tags = [
      { fieldValue: 'tag01', totalCount: 2 },
      { fieldValue: 'tag02', totalCount: 4 },
      { fieldValue: 'tag03', totalCount: 6 },
    ] as markdownRemarkGroupConnectionConnection[]

    const wrapper = shallow(<TagsCard tags={tags} Link={LinkStub} tag="tag01" />)

    expect(wrapper).toMatchSnapshot()
  })
})
