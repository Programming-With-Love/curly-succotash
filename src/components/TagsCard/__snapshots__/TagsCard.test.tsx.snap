// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsCard component should have on tag active 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    Link={[Function]}
    tag="tag01"
    tags={
        Array [
            Object {
              "fieldValue": "tag01",
              "totalCount": 2,
            },
            Object {
              "fieldValue": "tag02",
              "totalCount": 4,
            },
            Object {
              "fieldValue": "tag03",
              "totalCount": 6,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <CardContent>
          <CardHeader>
                    Tags
          </CardHeader>
</CardContent>,
        <CardContent>
          <List>
                    <ListItem
                              as="span"
                    >
                              <ListIcon
                                        color="blue"
                                        name="tag"
                              />
                              <ListContent
                                        style={
                                                  Object {
                                                            "fontWeight": "700",
                                                          }
                                        }
                              >
                                        <LinkStub
                                                  to="/blog"
                                        >
                                                  tag01
                                                   (
                                                  2
                                                  )
                                        </LinkStub>
                              </ListContent>
                    </ListItem>
                    <ListItem
                              as="span"
                    >
                              <ListIcon
                                        color={null}
                                        name="tag"
                              />
                              <ListContent
                                        style={null}
                              >
                                        <LinkStub
                                                  to="/blog/tags/tag02/"
                                        >
                                                  tag02
                                                   (
                                                  4
                                                  )
                                        </LinkStub>
                              </ListContent>
                    </ListItem>
                    <ListItem
                              as="span"
                    >
                              <ListIcon
                                        color={null}
                                        name="tag"
                              />
                              <ListContent
                                        style={null}
                              >
                                        <LinkStub
                                                  to="/blog/tags/tag03/"
                                        >
                                                  tag03
                                                   (
                                                  6
                                                  )
                                        </LinkStub>
                              </ListContent>
                    </ListItem>
          </List>
</CardContent>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <CardHeader>
            Tags
</CardHeader>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Tags",
          },
          "ref": null,
          "rendered": "Tags",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <List>
            <ListItem
                        as="span"
            >
                        <ListIcon
                                    color="blue"
                                    name="tag"
                        />
                        <ListContent
                                    style={
                                                Object {
                                                            "fontWeight": "700",
                                                          }
                                    }
                        >
                                    <LinkStub
                                                to="/blog"
                                    >
                                                tag01
                                                 (
                                                2
                                                )
                                    </LinkStub>
                        </ListContent>
            </ListItem>
            <ListItem
                        as="span"
            >
                        <ListIcon
                                    color={null}
                                    name="tag"
                        />
                        <ListContent
                                    style={null}
                        >
                                    <LinkStub
                                                to="/blog/tags/tag02/"
                                    >
                                                tag02
                                                 (
                                                4
                                                )
                                    </LinkStub>
                        </ListContent>
            </ListItem>
            <ListItem
                        as="span"
            >
                        <ListIcon
                                    color={null}
                                    name="tag"
                        />
                        <ListContent
                                    style={null}
                        >
                                    <LinkStub
                                                to="/blog/tags/tag03/"
                                    >
                                                tag03
                                                 (
                                                6
                                                )
                                    </LinkStub>
                        </ListContent>
            </ListItem>
</List>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <ListItem
                as="span"
>
                <ListIcon
                                color="blue"
                                name="tag"
                />
                <ListContent
                                style={
                                                Object {
                                                                "fontWeight": "700",
                                                              }
                                }
                >
                                <LinkStub
                                                to="/blog"
                                >
                                                tag01
                                                 (
                                                2
                                                )
                                </LinkStub>
                </ListContent>
</ListItem>,
              <ListItem
                as="span"
>
                <ListIcon
                                color={null}
                                name="tag"
                />
                <ListContent
                                style={null}
                >
                                <LinkStub
                                                to="/blog/tags/tag02/"
                                >
                                                tag02
                                                 (
                                                4
                                                )
                                </LinkStub>
                </ListContent>
</ListItem>,
              <ListItem
                as="span"
>
                <ListIcon
                                color={null}
                                name="tag"
                />
                <ListContent
                                style={null}
                >
                                <LinkStub
                                                to="/blog/tags/tag03/"
                                >
                                                tag03
                                                 (
                                                6
                                                )
                                </LinkStub>
                </ListContent>
</ListItem>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "tag01",
              "nodeType": "class",
              "props": Object {
                "as": "span",
                "children": Array [
                  <ListIcon
                    color="blue"
                    name="tag"
/>,
                  <ListContent
                    style={
                                        Object {
                                                            "fontWeight": "700",
                                                          }
                    }
>
                    <LinkStub
                                        to="/blog"
                    >
                                        tag01
                                         (
                                        2
                                        )
                    </LinkStub>
</ListContent>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "color": "blue",
                    "name": "tag",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <LinkStub
                      to="/blog"
>
                      tag01
                       (
                      2
                      )
</LinkStub>,
                    "style": Object {
                      "fontWeight": "700",
                    },
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        "tag01",
                        " (",
                        2,
                        ")",
                      ],
                      "to": "/blog",
                    },
                    "ref": null,
                    "rendered": Array [
                      "tag01",
                      " (",
                      2,
                      ")",
                    ],
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "tag02",
              "nodeType": "class",
              "props": Object {
                "as": "span",
                "children": Array [
                  <ListIcon
                    color={null}
                    name="tag"
/>,
                  <ListContent
                    style={null}
>
                    <LinkStub
                                        to="/blog/tags/tag02/"
                    >
                                        tag02
                                         (
                                        4
                                        )
                    </LinkStub>
</ListContent>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "color": null,
                    "name": "tag",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <LinkStub
                      to="/blog/tags/tag02/"
>
                      tag02
                       (
                      4
                      )
</LinkStub>,
                    "style": null,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        "tag02",
                        " (",
                        4,
                        ")",
                      ],
                      "to": "/blog/tags/tag02/",
                    },
                    "ref": null,
                    "rendered": Array [
                      "tag02",
                      " (",
                      4,
                      ")",
                    ],
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "tag03",
              "nodeType": "class",
              "props": Object {
                "as": "span",
                "children": Array [
                  <ListIcon
                    color={null}
                    name="tag"
/>,
                  <ListContent
                    style={null}
>
                    <LinkStub
                                        to="/blog/tags/tag03/"
                    >
                                        tag03
                                         (
                                        6
                                        )
                    </LinkStub>
</ListContent>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "color": null,
                    "name": "tag",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <LinkStub
                      to="/blog/tags/tag03/"
>
                      tag03
                       (
                      6
                      )
</LinkStub>,
                    "style": null,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        "tag03",
                        " (",
                        6,
                        ")",
                      ],
                      "to": "/blog/tags/tag03/",
                    },
                    "ref": null,
                    "rendered": Array [
                      "tag03",
                      " (",
                      6,
                      ")",
                    ],
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <CardContent>
            <CardHeader>
                        Tags
            </CardHeader>
</CardContent>,
          <CardContent>
            <List>
                        <ListItem
                                    as="span"
                        >
                                    <ListIcon
                                                color="blue"
                                                name="tag"
                                    />
                                    <ListContent
                                                style={
                                                            Object {
                                                                        "fontWeight": "700",
                                                                      }
                                                }
                                    >
                                                <LinkStub
                                                            to="/blog"
                                                >
                                                            tag01
                                                             (
                                                            2
                                                            )
                                                </LinkStub>
                                    </ListContent>
                        </ListItem>
                        <ListItem
                                    as="span"
                        >
                                    <ListIcon
                                                color={null}
                                                name="tag"
                                    />
                                    <ListContent
                                                style={null}
                                    >
                                                <LinkStub
                                                            to="/blog/tags/tag02/"
                                                >
                                                            tag02
                                                             (
                                                            4
                                                            )
                                                </LinkStub>
                                    </ListContent>
                        </ListItem>
                        <ListItem
                                    as="span"
                        >
                                    <ListIcon
                                                color={null}
                                                name="tag"
                                    />
                                    <ListContent
                                                style={null}
                                    >
                                                <LinkStub
                                                            to="/blog/tags/tag03/"
                                                >
                                                            tag03
                                                             (
                                                            6
                                                            )
                                                </LinkStub>
                                    </ListContent>
                        </ListItem>
            </List>
</CardContent>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <CardHeader>
              Tags
</CardHeader>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Tags",
            },
            "ref": null,
            "rendered": "Tags",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <List>
              <ListItem
                            as="span"
              >
                            <ListIcon
                                          color="blue"
                                          name="tag"
                            />
                            <ListContent
                                          style={
                                                        Object {
                                                                      "fontWeight": "700",
                                                                    }
                                          }
                            >
                                          <LinkStub
                                                        to="/blog"
                                          >
                                                        tag01
                                                         (
                                                        2
                                                        )
                                          </LinkStub>
                            </ListContent>
              </ListItem>
              <ListItem
                            as="span"
              >
                            <ListIcon
                                          color={null}
                                          name="tag"
                            />
                            <ListContent
                                          style={null}
                            >
                                          <LinkStub
                                                        to="/blog/tags/tag02/"
                                          >
                                                        tag02
                                                         (
                                                        4
                                                        )
                                          </LinkStub>
                            </ListContent>
              </ListItem>
              <ListItem
                            as="span"
              >
                            <ListIcon
                                          color={null}
                                          name="tag"
                            />
                            <ListContent
                                          style={null}
                            >
                                          <LinkStub
                                                        to="/blog/tags/tag03/"
                                          >
                                                        tag03
                                                         (
                                                        6
                                                        )
                                          </LinkStub>
                            </ListContent>
              </ListItem>
</List>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <ListItem
                  as="span"
>
                  <ListIcon
                                    color="blue"
                                    name="tag"
                  />
                  <ListContent
                                    style={
                                                      Object {
                                                                        "fontWeight": "700",
                                                                      }
                                    }
                  >
                                    <LinkStub
                                                      to="/blog"
                                    >
                                                      tag01
                                                       (
                                                      2
                                                      )
                                    </LinkStub>
                  </ListContent>
</ListItem>,
                <ListItem
                  as="span"
>
                  <ListIcon
                                    color={null}
                                    name="tag"
                  />
                  <ListContent
                                    style={null}
                  >
                                    <LinkStub
                                                      to="/blog/tags/tag02/"
                                    >
                                                      tag02
                                                       (
                                                      4
                                                      )
                                    </LinkStub>
                  </ListContent>
</ListItem>,
                <ListItem
                  as="span"
>
                  <ListIcon
                                    color={null}
                                    name="tag"
                  />
                  <ListContent
                                    style={null}
                  >
                                    <LinkStub
                                                      to="/blog/tags/tag03/"
                                    >
                                                      tag03
                                                       (
                                                      6
                                                      )
                                    </LinkStub>
                  </ListContent>
</ListItem>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "tag01",
                "nodeType": "class",
                "props": Object {
                  "as": "span",
                  "children": Array [
                    <ListIcon
                      color="blue"
                      name="tag"
/>,
                    <ListContent
                      style={
                                            Object {
                                                                  "fontWeight": "700",
                                                                }
                      }
>
                      <LinkStub
                                            to="/blog"
                      >
                                            tag01
                                             (
                                            2
                                            )
                      </LinkStub>
</ListContent>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "color": "blue",
                      "name": "tag",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <LinkStub
                        to="/blog"
>
                        tag01
                         (
                        2
                        )
</LinkStub>,
                      "style": Object {
                        "fontWeight": "700",
                      },
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": Array [
                          "tag01",
                          " (",
                          2,
                          ")",
                        ],
                        "to": "/blog",
                      },
                      "ref": null,
                      "rendered": Array [
                        "tag01",
                        " (",
                        2,
                        ")",
                      ],
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "tag02",
                "nodeType": "class",
                "props": Object {
                  "as": "span",
                  "children": Array [
                    <ListIcon
                      color={null}
                      name="tag"
/>,
                    <ListContent
                      style={null}
>
                      <LinkStub
                                            to="/blog/tags/tag02/"
                      >
                                            tag02
                                             (
                                            4
                                            )
                      </LinkStub>
</ListContent>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "color": null,
                      "name": "tag",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <LinkStub
                        to="/blog/tags/tag02/"
>
                        tag02
                         (
                        4
                        )
</LinkStub>,
                      "style": null,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": Array [
                          "tag02",
                          " (",
                          4,
                          ")",
                        ],
                        "to": "/blog/tags/tag02/",
                      },
                      "ref": null,
                      "rendered": Array [
                        "tag02",
                        " (",
                        4,
                        ")",
                      ],
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "tag03",
                "nodeType": "class",
                "props": Object {
                  "as": "span",
                  "children": Array [
                    <ListIcon
                      color={null}
                      name="tag"
/>,
                    <ListContent
                      style={null}
>
                      <LinkStub
                                            to="/blog/tags/tag03/"
                      >
                                            tag03
                                             (
                                            6
                                            )
                      </LinkStub>
</ListContent>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "color": null,
                      "name": "tag",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <LinkStub
                        to="/blog/tags/tag03/"
>
                        tag03
                         (
                        6
                        )
</LinkStub>,
                      "style": null,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": Array [
                          "tag03",
                          " (",
                          6,
                          ")",
                        ],
                        "to": "/blog/tags/tag03/",
                      },
                      "ref": null,
                      "rendered": Array [
                        "tag03",
                        " (",
                        6,
                        ")",
                      ],
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
